backends/p4tools/modules/testgen/targets/bmv2/test/TestTemplate.cmake

```diff
@@ -18,6 +18,51 @@ macro(check_with_bmv2 testfile testfolder p4test)
   file(APPEND ${testfile} "done\n")
 endmacro(check_with_bmv2)
 
+
```

Comment:
Can you make this a shell script? It will be easier to debug and read. You could also make this a Python script.